packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -235,10 +239,18 @@ export class ProjectService {
             return { warehouseClient: existingClient, sshTunnel };
         }
         // otherwise create a new client and cache for future use
+        const credentialsWithWarehouse =
+            credentials.type === WarehouseTypes.SNOWFLAKE
+                ? {
+                      ...warehouseSshCredentials,
+                      warehouse:
+                          snowflakeVirtualWarehouse || credentials.warehouse,
+                  }
+                : credentials;
```

Comment:
This is a bug, it should be `warehouseSshCredentials` which contains the credentials including the proxied host name and port for any active SSH connections.

Reply:
Before, if it wasn't snowflake it was returning `credentials` (i.e. the raw credentials) not `warehouseSshCredentials` (the proxied credentials)